prow/cmd/deck/main.go

```diff
@@ -1172,18 +1187,28 @@ func handleRerun(prowJobClient prowv1.ProwJobInterface, createProwJob bool) http
 		// On Prow instances that require auth even for viewing Deck this is okayish, because the Prowjob UUID
 		// is hard to guess
 		// Ref: https://github.com/kubernetes/test-infra/pull/12827#issuecomment-502850414
-		if createProwJob {
-			if r.Method != http.MethodPost {
-				http.Error(w, "request must be of type POST", http.StatusMethodNotAllowed)
-				return
-			}
-			if _, err := prowJobClient.Create(&newPJ); err != nil {
-				logrus.WithError(err).Error("Error creating job")
-				http.Error(w, fmt.Sprintf("Error creating job: %v", err), http.StatusInternalServerError)
-				return
+		if r.Method == http.MethodPost {
+			if createProwJob {
+				githubCookie, err := r.Cookie("github_login")
+				if err == http.ErrNoCookie {
+					http.Redirect(w, r, "/github-login", http.StatusFound)
+				} else if err != nil {
+					http.Error(w, fmt.Sprintf("Error retrieving GitHub cookie: %v", err), http.StatusInternalServerError)
+					return
+				}
+				if canTriggerJob(githubCookie.Value, cfg) {
+					if _, err := prowJobClient.Create(&newPJ); err != nil {
+						logrus.WithError(err).Error("Error creating job")
+						http.Error(w, fmt.Sprintf("Error creating job: %v", err), http.StatusInternalServerError)
+						return
+					}
+					http.Redirect(w, r, "/?rerun=success", http.StatusFound)
+					return
+				} else {
```

Comment:
Don't use else for this kind of thing. Correct golang style is to keep indentation low

So

```golang
if !canTriggerJob(gc.v, cfg) {
  http.Redirect(w, r, "/?rerun=denied", http.StatusFound)
  return
}
// create and redirect to success
```

Can you make sure you're always following this pattern?

https://dave.cheney.net/practical-go/presentations/qcon-china.html#_return_early_rather_than_nesting_deeply